### ISSUE TITLE
no gradients w.r.t. light object parameters

### ISSUE THREAD

**Issue by nikjetchev** [facebookresearch/pytorch3d#831]
I am trying to optimize a scene, including w.r.t. to the light parameters **lparams** , but their gradient is None.
Is this normal behaviour (pytorch3d does not support differentiable light sources) or am i doing something wrong?

code for light parametrization

```
lparams = torch.cuda.FloatTensor(12).uniform_(-0.05,0.05)
lparams=nn.Parameter(lparams)

def calcLights():
  l=PointLights(device=device, location=[[-3+lparams[0],0+lparams[1],0+lparams[2]]],ambient_color=((0.5+lparams[3], 0.5+lparams[4], 0.5+lparams[5]), ),\
                diffuse_color=((0.3+lparams[6], 0.3+lparams[7], 0.3+lparams[8]), ), specular_color=((0.2+lparams[9], 0.2+lparams[10], 0.2+lparams[11]), ))
  return l
```

snippet of overall rendering code

```
initL= PointLights(device=device, location=[[-3,0,0]])

sigma =1e-4*1e-2#*1e-2
raster_settings_soft = RasterizationSettings(
    image_size=224+112*0, 
    blur_radius=np.log(1. / 1e-4 - 1.)*sigma ,
    faces_per_pixel=1*12, perspective_correct=False
)

renderer = MeshRenderer(
    rasterizer=MeshRasterizer(
        cameras=camera, 
        raster_settings=raster_settings_soft
    ),
    shader=SoftPhongShader(
        device=device, 
        cameras=camera,
        lights=initL, materials=material
    )
)
#material some additional param
#mesh and camera additional parameters -- gradients of mesh works ok

lights = calcLights()
 images = renderer(mesh, lights=lights,cameras=camera)

loss = images.sum()#just some dummy loss for illustration
loss.backward()
```
After running this, the gradient of lparams is None
so how can I get loss differentiable w.r.t. the lparams, the light parameters?

thanks a lot for the help

**Comment by bottler** [comment]
I think this might be similar to #423. If you initialise the PointLights class with a tuple it creates a new tensor which breaks the gradient graph.

Can you try to replace 
```
def calcLights():
  l=PointLights(device=device, location=[[-3+lparams[0],0+lparams[1],0+lparams[2]]],ambient_color=((0.5+lparams[3], 0.5+lparams[4], 0.5+lparams[5]), ),\
                diffuse_color=((0.3+lparams[6], 0.3+lparams[7], 0.3+lparams[8]), ), specular_color=((0.2+lparams[9], 0.2+lparams[10], 0.2+lparams[11]), ))
  return l
```
with something like
```
location_offset = lparams.new_tensor([-3,0,0])
def calcLights():
    l = PointLights(
        device=device,
        location = lparams[None, :3][None] + location_offset,
        ambient_color=0.5 + lparams[None, 3:6],
        diffuse_color = 0.3 + lparams[None, 6:9],
        specular_color = 0.2+lparams[None, 9:],
        )
    return l
````
